### PR TITLE
Discovery Diffs

### PR DESCRIPTION
**Pull request by kickroot**
Omnibus MR for PROD-2636, which aligns Magpie discovery data shapes to that used in Open Raven production today.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -27,7 +27,7 @@
     <maven.compiler.target>11</maven.compiler.target>
     <jackson.version>2.12.1</jackson.version>
     <testcontainer.version>1.16.2</testcontainer.version>
-    <magpie-data.vesrion>0.1.10</magpie-data.vesrion>
+    <magpie-data.vesrion>0.1.11</magpie-data.vesrion>
```
Comment (mgurner1234) [inline pom.xml:30]: version is misspelt in the tag here

File: magpie-aws/src/main/java/io/openraven/magpie/plugins/aws/discovery/AWSUtils.java [hunk 2]
```diff
@@ -195,4 +197,27 @@ public static GetMetricStatisticsResponse getCloudwatchMetricStatistics( String
...
+      List<String> availableMetrics = new ArrayList<>();
+      List<DimensionFilter> dimensions = Collections.singletonList(DimensionFilter.builder().name("BucketName").value(bucketName).build());
+
+      final var request = ListMetricsRequest.builder()
+        .namespace("AWS/S3")
+        .metricName("BucketSizeBytes")
+        .dimensions(dimensions)
+        .build();
+      ListMetricsResponse response = client.listMetrics(request);
+      for (Metric metric : response.metrics()) {
```
Comment (mgurner1234) [inline magpie-aws/src/main/java/io/openraven/magpie/plugins/aws/discovery/AWSUtils.java:213]: Possible use of streams and collector here, but it doesn't read amazingly having tried:

      response.metrics()
        .forEach(metric -> metric.dimensions().stream().filter(dimension -> "StorageType".equals(dimension.name())).map(Dimension::value)
          .forEach(availableMetrics::add));

File: magpie-aws/src/main/java/io/openraven/magpie/plugins/aws/discovery/services/BackupDiscovery.java [hunk 3]
```diff
@@ -92,47 +95,68 @@ public void discoverPlans(ObjectMapper mapper, Session session, Region region, E
...
+
+  private void getPlanDetails(BackupClient client, MagpieAwsResource data) {
+    getAwsResponse(
+      () -> client.getBackupPlan(GetBackupPlanRequest.builder().backupPlanId(data.resourceId).build()),
+      (resp) -> AWSUtils.update(data.configuration, resp),
+      (noresp) -> AWSUtils.update(data.supplementaryConfiguration, noresp)
     );
   }
 
+//  private void discoverTags(BackupClient client, BackupVaultListMember resource, MagpieAwsResource data) {
```
Comment (mgurner1234) [inline magpie-aws/src/main/java/io/openraven/magpie/plugins/aws/discovery/services/BackupDiscovery.java:148]: Any reason to leave the commented block?

File: magpie-aws/src/main/java/io/openraven/magpie/plugins/aws/discovery/services/EC2Discovery.java [hunk 4]
```diff
@@ -88,7 +90,7 @@ public void discover(ObjectMapper mapper, Session session, Region region, Emitte
       discoverEIPs(mapper, session, client, region, emitter, account);
       discoverSecurityGroups(mapper, session, client, region, emitter, account, logger);
       discoverVolumes(mapper, session, client, region, emitter, account);
-      discoverSnapshots(mapper, session, client, region, emitter, account, logger);
+//      discoverSnapshots(mapper, session, client, region, emitter, account, logger);
```
Comment (mgurner1234) [inline magpie-aws/src/main/java/io/openraven/magpie/plugins/aws/discovery/services/EC2Discovery.java:93]: Any reason to leave comment?

File: magpie-aws/src/main/java/io/openraven/magpie/plugins/aws/discovery/services/S3Discovery.java [hunk 5]
```diff
@@ -32,11 +32,17 @@
 import io.openraven.magpie.plugins.aws.discovery.VersionedMagpieEnvelopeProvider;
 import org.javatuples.Pair;
 import org.slf4j.Logger;
+import software.amazon.awssdk.auth.credentials.AwsCredentialsProvider;
```
Comment (mgurner1234) [inline magpie-aws/src/main/java/io/openraven/magpie/plugins/aws/discovery/services/S3Discovery.java:35]: Unused imports left in this one

**Review by mgurner1234 — COMMENTED** [review]
Finished review, added a few non functional / minor comments.

File: pom.xml (hunk 1 above)
Comment (kickroot) [reply pom.xml:30]: 🤦

File: magpie-aws/src/main/java/io/openraven/magpie/plugins/aws/discovery/services/BackupDiscovery.java (hunk 3 above)
Comment (kickroot) [reply magpie-aws/src/main/java/io/openraven/magpie/plugins/aws/discovery/services/BackupDiscovery.java:148]: None at all!

File: magpie-aws/src/main/java/io/openraven/magpie/plugins/aws/discovery/AWSUtils.java (hunk 2 above)
Comment (kickroot) [reply magpie-aws/src/main/java/io/openraven/magpie/plugins/aws/discovery/AWSUtils.java:213]: I think with some line breaks and indentation your code reads better.

Thanks!

**Review by mgurner1234 — APPROVED** [review]

**Pull request merged**